core/src/test/java/io/questdb/griffin/UnionAllCastTest.java

```diff
@@ -1035,11 +1035,11 @@ public void testGeoLongGeoShort() throws Exception {
         // long + geohash overlap via string type
```

Comment:
There's something weird in geohash cast. 

```sql
create table x1 as (select rnd_geohash(40) a from long_sequence(5))
create table y1 as (select rnd_geohash(10) b from long_sequence(5))

select * from x1;
--w5qz4zkv
--uf5r0gp5
--0fj0e0cq
--7bq1kkdq
--2rhugbf2

selecT * from y1; 
--mt
--yr
--jn
--rw
--s0

selecT * from x1
union all
selecT * from y1
--kv
--gk
--71
--00
--p5
--mt
--yr
--jn
--rw
--s0

selecT *, cast(a as geohash(10b)) from x1
--w5qz4zkv w5
--uf5r0gp5 uf
--0fj0e0cq 0f
--7bq1kkdq 7b
--2rhugbf2 2r
```

x1 values cast to geohash(10) don't look like those in the final query .

Reply:
fixed in https://github.com/questdb/questdb/pull/2423/commits/d93c4b68350b07929dd4a4d84ec388948f9f1bfc

Reply:
why the last example is an issue?

![image](https://user-images.githubusercontent.com/7276403/192341466-bcccf26f-da89-40cf-952c-0421bddfb5a8.png)

Reply:
mind you, this example:
> create table x2 as (select x a, cast( x as geohash(2c) ) b from ( select rnd_geohash(40) x  from long_sequence(5) ) );

creates an unstable dataset (`rnd_geohash` is lazily invoked, twice, with random generator advancing each time)

![image](https://user-images.githubusercontent.com/7276403/192341927-792c06fa-43ad-44de-8b5f-e81cd72ce591.png)

Reply:
That's because it leads to :

```sql
select * from (select x a, x b  from ( select rnd_geohash(40) x from long_sequence(5) ) ) where a = b ;
--returns no result 
```

Compare to following sql in PG : 
```sql
select a, a from (select random() a from generate_Series(1,10,1)) as foo where a = a ;
--returns 10 lines with two identical columns 
```

So once rnd_geohash is evaluated in innermost query aliases should retain the value instead of behaving like separate rnd_geohash() calls .
I checked it behaves the same on demo/master so I'll create separate GH issue for it.

Reply:
you're right, but we don't have an API to "move row" on functions to stop them caching values. I can't see how we can implement it without major performance drag.